lib/check_data_points.rb

```diff
@@ -0,0 +1,76 @@
+class CheckDataPoints
+  # 1) Number of tipline messages
+  def self.tipline_messages(team_id, start_date, end_date, granularity = nil)
+    start_date, end_date = self.parse_start_end_dates(start_date, end_date)
+    query = TiplineMessage.where(team_id: team_id, created_at: start_date..end_date)
+    self.query_based_on_granularity(query, granularity)
+  end
+
+  # 2) Number of tipline requests
+  def self.tipline_requests(team_id, start_date, end_date, granularity = nil)
+    start_date, end_date = self.parse_start_end_dates(start_date, end_date)
+    query = TiplineRequest.where(team_id: team_id, created_at: start_date..end_date)
+    self.query_based_on_granularity(query, granularity)
+  end
+
+  # 3) Number of tipline requests grouped by type of search result
+  def self.tipline_requests_by_search_type(team_id, start_date, end_date)
+    start_date, end_date = self.parse_start_end_dates(start_date, end_date)
+    TiplineRequest.where(
+      team_id: team_id,
+      smooch_request_type: ['relevant_search_result_requests', 'irrelevant_search_result_requests'],
```

Comment:
Let's add the timeout search result case too (even if the frontend doesn't use it).

Reply:
Done